jraft-core/src/test/java/com/alipay/sofa/jraft/util/BytesUtilTest.java

```diff
@@ -0,0 +1,78 @@
+package com.alipay.sofa.jraft.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class BytesUtilTest {
+
+    @Test
+    public void testNullToEmpty() {
+        Assert.assertArrayEquals(new byte[]{}, BytesUtil.nullToEmpty(null));
+        Assert.assertArrayEquals(new byte[]{1, 2}, BytesUtil.nullToEmpty(new byte[]{1, 2}));
+    }
+
+    @Test
+    public void testIsEmpty() {
+        Assert.assertTrue(BytesUtil.isEmpty(null));
+
+        Assert.assertFalse(BytesUtil.isEmpty(new byte[]{1, 2}));
+    }
+
+    @Test
+    public void testWriteUtf8() {
+        Assert.assertNull(BytesUtil.writeUtf8(null));
+
+        Assert.assertArrayEquals(new byte[]{102, 111, 111}, BytesUtil.writeUtf8("foo"));
+    }
+
+    @Test
+    public void testReadUtf8() {
+        Assert.assertNull(BytesUtil.readUtf8(null));
+
+        Assert.assertEquals("foo", BytesUtil.readUtf8(new byte[]{102, 111, 111}));
+    }
+
+    @Test
+    public void testNextBytes() {
+        Assert.assertArrayEquals(new byte[]{0}, BytesUtil.nextBytes(new byte[]{}));
+        Assert.assertArrayEquals(new byte[]{1, 2, 0}, BytesUtil.nextBytes(new byte[]{1, 2}));
+    }
+
+    @Test
+    public void testCompare() {
+        byte[] array = new byte[]{1, 2};
+
+        Assert.assertEquals(0, BytesUtil.compare(array, array));
+        Assert.assertEquals(-2, BytesUtil.compare(new byte[]{1, 2}, new byte[]{3, 4}));
+        Assert.assertEquals(0, BytesUtil.compare(new byte[]{3, 4}, new byte[]{3, 4}));
+    }
+
+    @Test
+    public void testMax() {
+        byte[] array = new byte[]{3, 4};
+
+        Assert.assertArrayEquals(array, BytesUtil.max(array, array));
+        Assert.assertArrayEquals(array, BytesUtil.max(new byte[]{1, 2}, array));
```

Comment:
I might be wrong but I'm not sure what this is actually testing? I would expect the assertion to detect what the `max` result is? Same with `min` below.

Reply:
It compares which array has the larger values and returns that array. Same for min method